common-lib/common/panels/all/base.libsonnet

```diff
@@ -0,0 +1,13 @@
+local g = import 'g.libsonnet';
+
+local timeSeries = g.panel.timeSeries;
+local fieldOverride = g.panel.timeSeries.fieldOverride;
+local custom = timeSeries.fieldConfig.defaults.custom;
+local defaults = timeSeries.fieldConfig.defaults;
+local options = timeSeries.options;
+{
+  new(title, targets, description=''):
+    timeSeries.queryOptions.withInterval('1m')
+    + timeSeries.queryOptions.withTargets(targets)
+    + timeSeries.panelOptions.withDescription(description),
```

Comment:
These feel out of place to me. Tho I may be struggling with the grafonnet object model more than anything.

It seems like we're applying these properties to the entire panel, where they should be applied to each query/target individually.

Why are we dictating 1m interval? I suppose the purpose of this library is to be opinionated. 🤔 If we *do* set an interval for an entire panel, it should be a `Mixin` type function that recursively and retroactively applies to all queries/targets which have been added.

Reply:
This is a tricky part.
1. If interval is not defined at all, it will be sort of 'none' because panel's common datasource is 'Mixed'. Setting it one 1m here, is sort of hack, that i'll try to address.

Reply:
Ahh, I see! I think I experienced this on my hackathon project where my timeseries panel was not rendering or behaving as expected. This was very likely the cause.

Maybe having interval as an optional parameter, with documentation describing the reason it needs to be set (as you described above) would be the best approach.

Reply:
this one solved, see logic in panels/generic/base (tries to set panel's datasource based on target's datasource)